python/ray/serve/_private/proxy.py

```diff
@@ -1293,6 +1294,10 @@ def _get_logging_config(self) -> Tuple:
                 log_file_path = handler.baseFilename
         return log_file_path
 
+    def _dump_ingress_replicas_for_testing(self, route) -> Set[ReplicaID]:
```

Comment:
non-blocker: I kinda feel the way this is currently used, we don't really need to filter the replica by the route. We can probably just directly using `_dump_replica_states_for_testing()` from the controller? 

But also know that this is just refactored from the existing test, it's totally fine by me

Reply:
nit: 
```suggestion
    def _dump_ingress_replicas_for_testing(self, route: str) -> Set[ReplicaID]:
```

Reply:
`_dump_replica_states_for_testing` from the controller will give the list of replicas that the controller sees, but to deflake the test we need to make sure the replicas have been broadcasted to the proxy.